modules/gitui/hm.nix

```
@@ -0,0 +1,43 @@
+{ config, lib, ... }:
+let
+  mkRgb = color:
+    let
+      c = config.lib.stylix.colors;
+      r = c."${color}-rgb-r";
+      g = c."${color}-rgb-g";
+      b = c."${color}-rgb-b";
+    in
+    "Rgb(${r}, ${g}, ${b})";
```

Unless Nix efficiently caches the `c = config.lib.stylix.colors;` lookup, it would be more efficient to explicitly cache it without relying on magical optimizations:

```suggestion
  mkRgb = let
    c = config.lib.stylix.colors;
  in
    color:
      let
        r = c."${color}-rgb-r";
        g = c."${color}-rgb-g";
        b = c."${color}-rgb-b";
      in
        "Rgb(${r}, ${g}, ${b})";
```

Personally, I would avoid one-letter variables, although `r`, `g`, and `b` are probably fine in this context:

```suggestion
  mkRgb = let
    colors = config.lib.stylix.colors;
  in
    color:
      let
        r = colors."${color}-rgb-r";
        g = colors."${color}-rgb-g";
        b = colors."${color}-rgb-b";
      in
        "Rgb(${r}, ${g}, ${b})";
```

---

I can change that no problem, I copied this off [the rofi module](https://github.com/danth/stylix/blob/43d23b1609b87f6a4100db2a09bd118c52c78766/modules/rofi/hm.nix#L9)